### PR TITLE
Add standard name metadata to most EAMxx output variables

### LINKED ISSUE #700: [AD] Implement CF-Compliance Checking in Field Repo
Implement the infrastructure to check all newly registered variables in the field manager against the CF compliant dictionaries. Note that PR #757 makes the first step towards this goal by creating a _standalone_ tool that checks if a name is in the CF database but does not implement _online_ checking.

As a developer tool, establish a script or executable which can parse the CF compliant dictionaries and return possible long-name strings given a set of inputs. For example, if the developer wants a variable and isn't sure what the name should be, but does know the units are in Pa, query for all variables with unit=Pa.  Or that the variable is related to pressure, sure for instances of the string "pres" or "pressure".

This is step 2 of getting CF compliance in SCREAM.

Addresses #527 , #528 and  #699

### PR DESCRIPTION
**Pull request by AaronDonahue**
This commit brings EAMxx into CF-Compliance by adding `standard_name` metadata for a subset of the EAMxx output variables that have one.

Fixes #700

### CODE REVIEW

File: components/eamxx/src/share/io/scream_io_utils.hpp [hunk 1]
```diff
@@ -91,6 +100,79 @@ struct LongNames {
...
+  std::map<std::string,std::string> name_2_standardname = {
+    {"p_mid"                                                       , "air_pressure"},
+    {"p_mid_at_cldtop"                                             , "air_pressure_at_cloud_top"},
+    {"T_2m"                                                        , "air_temperature"},
+    {"T_mid"                                                       , "air_temperature"},
+    {"T_mid_at_cldtop"                                             , "air_temperature_at_cloud_top"},
+    {"aero_g_sw"                                                   , "asymmetry_factor_of_ambient_aerosol_particles"},
+    {"pbl_height"                                                  , "atmosphere_boundary_layer_thickness"},
+    {"precip_liq_surf_mass"                                        , "atmosphere_mass_content_of_liquid_precipitation"},
+    {"cldlow"                                                      , "cloud_area_fraction"},
```
Comment (bartgol) [inline components/eamxx/src/share/io/scream_io_utils.hpp:114]: Are these really all the same?

**Review by bartgol — DISMISSED** [review]

File: components/eamxx/src/share/io/scream_io_utils.hpp (hunk 1 above)
Comment (AaronDonahue) [reply components/eamxx/src/share/io/scream_io_utils.hpp:114]: Yeah, I'm not sure.  I'm pretty sure I only grabbed standard_names I was able to confirm either on the CF website or from EAM.  Let me check again.

Comment (AaronDonahue) [reply components/eamxx/src/share/io/scream_io_utils.hpp]: Good catch, I found better standard names with a little more digging.

**Comment by AaronDonahue** [comment]
> Unsolicited review: wouldn't we prefer to have this dict-type info in a yaml-like file somewhere in this folder (or elsewhere) instead of having it be part of the cpp code? I believe we could then just read the yaml file (or similar) to construct the map.
> 
> Of course, we can do that later, not necessarily in this PR ... I think it is tidier in the long run to have info like this stored in standalone files ...

I like this idea.  I could probably make that a part of this PR.  @bartgol , do you have any thoughts on this approach?

**Review by bartgol — APPROVED** [review]
Looks good. Maybe we could change "long_name" to "description", since having a short, a long, and a standard name seem too much...

**Comment by bartgol** [comment]
> Unsolicited review: wouldn't we prefer to have this dict-type info in a yaml-like file somewhere in this folder (or elsewhere) instead of having it be part of the cpp code? I believe we could then just read the yaml file (or similar) to construct the map.
> 
> Of course, we can do that later, not necessarily in this PR ... I think it is tidier in the long run to have info like this stored in standalone files ...

@jeff-cohere and I were working a while ago on a tool to enable this. You can see we have two databases in `eamxx/data`, containing yaml files for field names. We also developed string-matching tools in ekat, so that users can query the database to find what the real name is. We also thought about using it at runtime, but never had time to implement a robust approach. We can try in a following PR. Are you volunteering? ;-)

**Review by bartgol — APPROVED** [review]

**Comment by bartgol** [comment]
It appears that mergify does indeed that: so long as the PR was approved "at some point", that check passed. There's a mergify action that allows to dismiss reviews though. I want to see if we can use it to dismiss stale reviews. If not, then byebye mergify

**Comment by rljacob** [comment]
I think there's a github branch protection setting that might help:  "Dismiss stale pull request approvals when new commits are pushed. New reviewable commits pushed to a matching branch will dismiss pull request review approvals."

**Pull request merged**